community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/UpdateGraph.scala

```diff
@@ -269,9 +269,19 @@ trait UpdateGraph {
   def deleteOverlap(qg: QueryGraph): Boolean = {
     // TODO:H FIXME qg.argumentIds here is not correct, but there is a unit test that depends on it
     val identifiersToRead = qg.allPatternNodesRead ++ qg.allPatternRelationshipsRead.map(_.name) ++ qg.argumentIds
-    (identifiersToRead intersect identifiersToDelete).nonEmpty
+    (identifiersToRead intersect identifiersToDelete).nonEmpty || deletedRelationshipsOverlap(qg)
   }
 
+  private def deletedRelationshipsOverlap(qg: QueryGraph): Boolean = {
+    val relsToRead = qg.allPatternRelationshipsRead.map(_.name)
+    val relDeleted = patternRelationships.filter { p =>
+      identifiersToDelete.contains(p.name)
+    }
+    relsToRead.nonEmpty && relDeleted.nonEmpty
```

Comment:
I think this hack is not comprehensive. A relationship identifier that is being deleted could possibly come from another preceding query graph. What we want is a sure way to decide if the identifier is a relationship. If we had the LogicalPlanningContext we could get it from context.semanticTable.isRelationship(x).